docs/milestone-1.md

```diff
@@ -0,0 +1,175 @@
+
+## API Docs
+
+Methods are prefixed using the following convention:
+- `engine` prefix indicates calls made to the execution client. These methods are specified in the [execution engine APIs](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md).
+- `builder` prefix indicates calls made to the mev-boost middleware.
+- `relay` prefix indicates calls made to a relay.
+
+### engine_forkchoiceUpdatedV1
+
+See [engine_forkchoiceUpdatedV1](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md#engine_forkchoiceupdatedv1).
+
+### engine_getPayloadV1
+
+See [engine_getPayloadV1](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md#engine_getpayloadv1).
+
+### builder_getPayloadHeaderV1
+
+#### Request
+
+- method: `builder_getPayloadHeaderV1`
+- params:
+  1. `payloadId`: `DATA`, 8 Bytes - Identifier of the payload build process
+
+#### Response
+
+- result: [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayloadheader)
+- error: code and message set in case an exception happens while getting the payload.
+
+### builder_proposeBlindedBlockV1
+
+#### Request
+
+- method: `builder_proposeBlindedBlockV1`
+- params:
+  1. [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock)
+
+#### Response
+
+- result: [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayload)
+- error: code and message set in case an exception happens while proposing the payload.
+
+Technically, this call only needs to return the `transactions` field of [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayload), but we return the full payload for simplicity.
+
+### relay_getPayloadHeaderV1
+
+#### Request
+
+- method: `relay_getPayloadHeaderV1`
+- params:
+  1. `payloadId`: `DATA`, 8 Bytes - Identifier of the payload build process
+
+#### Response
+
+- result: [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
+- error: code and message set in case an exception happens while getting the payload.
+
+### relay_proposeBlindedBlockV1
+
+#### Request
+
+- method: `relay_proposeBlindedBlockV1`
+- params:
+  1. [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock)
+  2. [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
+
+#### Response
+
+- result: [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayload)
+- error: code and message set in case an exception happens while proposing the payload.
+
+Technically, this call only needs to return the `transactions` field of [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayload), but we return the full payload for simplicity. 
+
+### Types
+
+#### SignedMEVPayloadHeader
```

Comment:
Along the lines of the comments about `SignedBlindedBeaconBlock`, but here, the JSON-RPC and SSZ types are freely intermixed, with the sort-of-SSZ-type-looking `SignedMEVPayloadHeader` containing a `MEVPayloadHeader` the fields of which are a `Data` and `Quantity` (JSON-RPC serialization types), which itself contains a `ExecutionPayloadHeaderV1`(a consensus layer SSZ type, defined using the SSZ-serialization-defined `Hash32`, `ExecutionAddress`, `Bytes32`, `uint64`, etc types). It's not internally consistent.

Implementing this might involve drawing out, at each layer, the dual JSON-RPC form of `SignedMEVPayloadHeader`, the dual SSZ form of `MEVPayloadHeader` (because the sort-of-SSZ-defined `SignedMEVPayloadHeader` either has to be read as a purely JSON-RPC object, or if not, for all its component parts to have consensus layer-like definitions in addition to the purely JSON-RPC definitions here), and likewise but inverted for the nominally-natively-SSZ `ExecutionPayloadHeaderV1`.

For example, one might imagine the CL/SSZ `MEVPayloadHeader`:
```python
class MEVPayloadHeader(Container):
    payloadHeader: ExecutionPayloadHeaderV1
    feeRecipient: ExecutionAddress
    feeRecipientBalance: uint256
```

https://github.com/realbigsean/lighthouse/blob/mev-lighthouse/beacon_node/execution_layer/src/engine_api/json_structures.rs fills this out with some gaps, but none of it is directly specified and there are various points of ambiguity.

One challenge

Reply:
oh what a mess. As I said before, I would like everything to be SSZ.

Reply:
Moved to https://github.com/flashbots/mev-boost/issues/76 for traceability and better discussion.